registry/registry.go

```diff
@@ -38,24 +40,24 @@ type Registry struct {
  * This passes http.DefaultTransport to WrapTransport when creating the
  * http.Client.
  */
-func New(registryUrl, username, password string) (*Registry, error) {
+func New(ctx context.Context, registryUrl, username, password string) (*Registry, error) {
```

Comment:
Ok that's pretty useful. It also implicitly tests auth credentials right (because apprunner is private)?